domain/service/servicedetails_store.go

```diff
@@ -142,6 +143,56 @@ func (s *storeImpl) GetServiceDetailsByParentID(ctx datastore.Context, parentID
 	return details, nil
 }
 
+// GetServiceDetailsByIDOrName returns the service details for any services
+// whose serviceID matches the query exactly or whose names contain the query
+// as a substring
+func GetServiceDetailsByIDOrName(ctx datastore.Context, query string) ([]ServiceDetails, error) {
```

Comment:
Please add the following so that we can measure this method:
```
defer ctx.Metrics().Stop(ctx.Metrics().Start("ServiceStore. GetServiceDetailsByIDOrName"))
```